src/Compiler.php

```diff
@@ -167,27 +161,41 @@ public function repr($value)
     /**
      * @return $this
      */
-    public function addDebugInfo(Node $node)
-    {
-        if ($node->getTemplateLine() != $this->lastLine) {
-            $this->write(sprintf("// line %d\n", $node->getTemplateLine()));
+	public function addDebugInfo(Node $node)
+	{
+    	$this->sourceLine = $node->getTemplateLine();
 
-            $this->sourceLine += substr_count($this->source, "\n", $this->sourceOffset);
-            $this->sourceOffset = \strlen($this->source);
-            $this->debugInfo[$this->sourceLine] = $node->getTemplateLine();
+    	$this->write("// line {$node->getTemplateLine()}\n");
 
-            $this->lastLine = $node->getTemplateLine();
-        }
+    	$templateLines = substr_count($node->getSource(), PHP_EOL);
+    	$this->sourceLine += $templateLines;
+    	$this->sourceOffset = strlen($this->source);
 
-        return $this;
-    }
+    	$this->source .= $node->getSource();
+
+    	$this->debugInfo[$node->getTemplateLine()] = $node->getTemplateLine();
+
+    	return $this;
+	}
 
     public function getDebugInfo(): array
     {
-        ksort($this->debugInfo);
+        \ksort($this->debugInfo);
 
-        return $this->debugInfo;
+        return $this->debugInfo ?? [];
     }
+    
+    function sortData($data)
```

Comment:
Where is it used ?

Reply:
Has been removed again and should not appear at all!